crates/red_knot_test/src/lib.rs

```diff
@@ -65,28 +71,43 @@ fn run_test(test: &parser::MarkdownTest) -> Result<(), Failures> {
         );
         let full_path = workspace_root.join(file.path);
         db.write_file(&full_path, file.code).unwrap();
-        system_paths.push(full_path);
+        paths.push(AbsoluteLineNumberPath {
+            path: full_path,
+            md_offset: file.md_offset,
+        });
     }
 
-    let mut failures = BTreeMap::default();
+    let mut failures = Vec::with_capacity(paths.len());
```

Comment:
This may be a holdover from a previous version of the PR, but I don't think this `with_capacity` really makes sense anymore. This vector isn't going to have a number of entries based on the number of embedded files, it'll have a number of entries based on the number of lines with failures on them. In the ideal scenario (where arguably we care the most about speed) it'll be an empty vec. So I think it would be better to optimistically not allocate:
```suggestion
    let mut failures = Vec::default();
```